src/ConnectionFactory.php

```diff
@@ -0,0 +1,65 @@
+<?php
+namespace Cadre\AtlasOrmDebugBarBridge;
+
+use Aura\Sql\ExtendedPdo as BaseExtendedPdo;
+use DebugBar\DataCollector\PDO\TraceablePDO;
+use PDO;
+
+class ConnectionFactory
+{
+    private $dsn;
+    private $username;
+    private $password;
+    private $options = array();
+    private $attributes = array();
+    private $pdo;
+
+    public function __construct(
+        $dsn,
+        $username = null,
+        $password = null,
+        array $options = array(),
+        array $attributes = array()
+    ) {
+        if ($dsn instanceof ExtendedPdo) {
+            $this->pdo = $dsn;
+        } elseif ($dsn instanceof BaseExtendedPdo) {
+            $this->pdo = new ExtendedPdo(new TraceablePDO($dsn->getPdo()));
+        } elseif ($dsn instanceof TraceablePDO) {
+            $this->pdo = new ExtendedPdo($dsn);
+        } elseif ($dsn instanceof PDO) {
+            $this->pdo = new ExtendedPdo(new TraceablePDO($dsn));
+        } else {
+            $this->dsn = $dsn;
```

Comment:
Why not create the instance here ?

Reason : It is any how lazy loaded. Also I don't know if we really need these much of configuration option. But I know different people, different configuration. But a bit complex.